src/resources/stores.jl

```diff
@@ -21,43 +20,47 @@ See [`Container`](@ref) for a more lock-like resource.
 
 Think of `Resource` and `Container` as locks and of `Store` as channels. They block only if empty (on taking) or full (on storing).
 """
-mutable struct Store{T} <: AbstractResource
+mutable struct Store{N, T<:Number} <: AbstractResource
   env :: Environment
   capacity :: UInt
   load :: UInt
-  items :: Dict{T, UInt}
+  items :: Dict{N, UInt}
   seid :: UInt
-  put_queue :: DataStructures.PriorityQueue{Put, StorePutKey{T}}
-  get_queue :: DataStructures.PriorityQueue{Get, StoreGetKey}
-  function Store{T}(env::Environment; capacity=typemax(UInt)) where {T}
-    new(env, UInt(capacity), zero(UInt), Dict{T, UInt}(), zero(UInt), DataStructures.PriorityQueue{Put, StorePutKey{T}}(), DataStructures.PriorityQueue{Get, StoreGetKey}())
+  put_queue :: DataStructures.PriorityQueue{Put, StorePutKey{N, T}}
+  get_queue :: DataStructures.PriorityQueue{Get, StoreGetKey{T}}
+  function Store{N, T}(env::Environment; capacity::UInt=typemax(UInt)) where {N, T<:Number}
+    new(env, capacity, zero(UInt), Dict{N, UInt}(), zero(UInt), DataStructures.PriorityQueue{Put, StorePutKey{N, T}}(), DataStructures.PriorityQueue{Get, StoreGetKey{T}}())
   end
 end
 
+function Store{N}(env::Environment; capacity::UInt=typemax(UInt)) where {N}
```

Comment:
I would suggest writing this as `capacity=typemax(UInt)` (without a type signature) and then in the next line casting `UInt(capacity)` to preserve the previously existing API.

Reply:
ok, coming in the next commit